### DISCUSSION TITLE
OPFOR Spawns

### DISCUSSION THREAD

**Discussion by Red-Dog-Actual** [Vdauphin/HeartsAndMinds#1435] [Question and answer] [answered]
I've noticed that there are an inordinate number of officers and NCOs spawning. Often half of the opfor spawning are officers and NCOs. Is there any way to change the chances of each unit class spawning? or a way to designate which unit classes are officers or NCOs?

Using a custom made faction if it matters.

**Comment by Vdauphin** [comment, marked as answer]
Hello,

> s there any way to change the chances of each unit class spawning?

No it is random

You can blacklist some classname here: https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/mission.sqf#L613

Cheers

**Reply by Red-Dog-Actual** [reply]
I know about blacklisting, but I was hoping there was a way to designate which classes were squad leaders or something. 

Maybe in the future you could add a way to designate class names for squad leaders, heavy weapons, medics, etc, and have them spawn in realistic proportions. 

I guess I'll just modify my faction to include more basic units to increase their chances of spawning more.

Love H&M by the way, definitely the most immersive counter insurgency mission in Arma.
